# <u>**Streaming Service Comparison**</u>

### **Objective:**
- Determine which streaming platform hosts the majority of content I enjoy so that I can pare down the services to which I subscribe. 

### **Data Sources:**
- [Netflix via Kaggle.com](https://www.kaggle.com/datasets/octopusteam/full-netflix-dataset)
- [Hulu via Kaggle.com](https://www.kaggle.com/datasets/octopusteam/full-hulu-dataset)
- [Prime via Kaggle.com](https://www.kaggle.com/datasets/octopusteam/full-amazon-prime-dataset/data)
- [AppleTV via Kaggle.com](https://www.kaggle.com/datasets/octopusteam/full-apple-tv-dataset)

## **Data Collection & Loading**

### **Import Pandas, Numpy, Matplotlib, Wordcloud, and PIL**

In [1]:
import glob
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import wordcloud as wc
from PIL import Image

### **Data Load**

In [2]:
# Load in each file separately. 
# Plan to create a function for this or NOT
apple = pd.read_csv("AppleTV.csv")
hulu = pd.read_csv("Hulu.csv")
netflix = pd.read_csv("Netflix.csv")
prime = pd.read_csv("Prime.csv")

### **Apple**   

In [3]:
#display(apple)
# apple.head(3)
# hulu.head(3)
# netflix.head(3)
# prime.head(3)
#display(apple)
print(apple.head(3))
print(hulu.head(3))
print(netflix.head(3))
print(prime.head(3))

             title   type          genres  releaseYear     imdbId  \
0       Four Rooms  movie          Comedy       1995.0  tt0113101   
1     Forrest Gump  movie  Drama, Romance       1994.0  tt0109830   
2  American Beauty  movie           Drama       1999.0  tt0169547   

   imdbAverageRating  imdbNumVotes availableCountries  
0                6.7      113403.0                NaN  
1                8.8     2348885.0                NaN  
2                8.3     1238903.0                NaN  
                 title   type                  genres  releaseYear     imdbId  \
0                Ariel  movie  Comedy, Crime, Romance       1988.0  tt0094675   
1  Shadows in Paradise  movie    Comedy, Drama, Music       1986.0  tt0092149   
2         Forrest Gump  movie          Drama, Romance       1994.0  tt0109830   

   imdbAverageRating  imdbNumVotes availableCountries  
0                7.4        8949.0                NaN  
1                7.5        7727.0                NaN  
2     

In [4]:
# Dispaly information about the dataset (i.e., row counts, column counts, column names, datatypes, # of non-null rows)
apple.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18208 entries, 0 to 18207
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   title               17643 non-null  object 
 1   type                18208 non-null  object 
 2   genres              17549 non-null  object 
 3   releaseYear         18179 non-null  float64
 4   imdbId              16727 non-null  object 
 5   imdbAverageRating   16314 non-null  float64
 6   imdbNumVotes        16314 non-null  float64
 7   availableCountries  84 non-null     object 
dtypes: float64(3), object(5)
memory usage: 1.1+ MB


In [5]:
apple.describe()

releaseYear  imdbAverageRating  imdbNumVotes
count  18179.000000       16314.000000  1.631400e+04
mean    2006.822432           6.381574  2.554855e+04
std       18.410460           1.162677  1.008288e+05
min     1902.000000           1.300000  5.000000e+00
25%     2001.000000           5.700000  2.010000e+02
50%     2014.000000           6.500000  1.234000e+03
75%     2020.000000           7.200000  8.106500e+03
max     2025.000000           9.600000  2.348885e+06

In [6]:
apple.select_dtypes("object").describe()

title   type genres     imdbId  \
count                          17643  18208  17549      16727   
unique                         16986      2    798      16724   
top     Teenage Mutant Ninja Turtles  movie  Drama  tt0112120   
freq                               5  14005   1590          2   

       availableCountries  
count                  84  
unique                 14  
top                    US  
freq                   42

In [7]:
# Drop availableCountries Column - no value with such a low non-null number
apple.drop("availableCountries",axis=1,inplace=True)
apple.columns

Index(['title', 'type', 'genres', 'releaseYear', 'imdbId', 'imdbAverageRating',
       'imdbNumVotes'],
      dtype='object')

In [8]:
# Change releaseYear,  imdbNumVotes to integers 
# Display only the data types

apple[["releaseYear","imdbNumVotes"]] = apple[["releaseYear","imdbNumVotes"]].apply(np.int64)
display(apple.dtypes)

c:\Users\rkynhoff\OneDrive - University of Rochester\Documents\CodeLouisville\Streaming_Service_Comparisons\venv\Lib\site-packages\pandas\core\series.py:1031: RuntimeWarning: invalid value encountered in cast
  arr = np.asarray(values, dtype=dtype)


title                 object
type                  object
genres                object
releaseYear            int64
imdbId                object
imdbAverageRating    float64
imdbNumVotes           int64
dtype: object

In [9]:
# Rename Columns
apple.rename(columns={"title":"Title","type":"Type","genres":"Combined Genres","releaseYear":"Release Year","imdbId":"IMDb ID","imdbAverageRating":"IMDb Average Rating","imdbNumVotes":"IMDb Num Votes"},inplace = True)
# Add a column to hulu DF and entered a default value for each row & display bottom two rows
apple["Service Name"] = "AppleTV"
apple.columns

Index(['Title', 'Type', 'Combined Genres', 'Release Year', 'IMDb ID',
       'IMDb Average Rating', 'IMDb Num Votes', 'Service Name'],
      dtype='object')

In [10]:
# Determine how many titles contain null values
apple["Title"].isna().sum()

np.int64(565)

In [11]:
# Drop the rows contiaining null values in the Title column
apple = apple.dropna(subset=["Title"])
apple["Title"].isna().sum()


np.int64(0)

In [12]:
apple.duplicated().value_counts()

False    17641
True         2
Name: count, dtype: int64

In [13]:
apple[apple.duplicated(keep=False)]

Title   Type  \
3497   A Personal Journey with Martin Scorsese Throug...  movie   
13091                The Kidnap & Murder of Lynda Spence  movie   
13093                The Kidnap & Murder of Lynda Spence  movie   
13980  A Personal Journey with Martin Scorsese Throug...  movie   

                       Combined Genres  Release Year    IMDb ID  \
3497   Biography, Documentary, History          1995  tt0112120   
13091               Documentary, Crime          2023        NaN   
13093               Documentary, Crime          2023        NaN   
13980  Biography, Documentary, History          1995  tt0112120   

       IMDb Average Rating       IMDb Num Votes Service Name  
3497                   8.5                 5196      AppleTV  
13091                  NaN -9223372036854775808      AppleTV  
13093                  NaN -9223372036854775808      AppleTV  
13980                  8.5                 5196      AppleTV

In [14]:
# Drop duplicate rows
apple.drop_duplicates()


Title   Type     Combined Genres  \
0                                Four Rooms  movie              Comedy   
1                              Forrest Gump  movie      Drama, Romance   
2                           American Beauty  movie               Drama   
3                              Citizen Kane  movie      Drama, Mystery   
4                                Metropolis  movie       Drama, Sci-Fi   
...                                     ...    ...                 ...   
18188     Das Boot - Die komplette TV-Serie     tv          Drama, War   
18194              The Amazon Review Killer     tv         Documentary   
18198                          Conspirators     tv             Mystery   
18200                   Extraordinary World     tv         Documentary   
18203  They Took Our Child: We Got Her Back     tv  Crime, Documentary   

       Release Year     IMDb ID  IMDb Average Rating       IMDb Num Votes  \
0              1995   tt0113101                  6.7               113403   
1              1994   tt0109830                  8.8              2348885   
2              1999   tt0169547                  8.3              1238903   
3              1941   tt0033467                  8.3               477481   
4              1927   tt0017136                  8.3               192285   
...             ...         ...                  ...                  ...   
18188          1985  tt30970892                  8.8                  198   
18194          2024  tt33342317                  6.4                   64   
18198          2025  tt35075008                  5.5                   44   
18200          2025  tt34888944                  NaN -9223372036854775808   
18203          2015   tt5056408                  8.6                   31   

      Service Name  
0          AppleTV  
1          AppleTV  
2          AppleTV  
3          AppleTV  
4          AppleTV  
...            ...  
18188      AppleTV  
18194      AppleTV  
18198      AppleTV  
18200      AppleTV  
18203      AppleTV  

[17641 rows x 8 columns]

In [15]:
apple[apple.duplicated(subset=["IMDb ID","Title","Release Year"], keep=False)]


Title   Type  \
3497   A Personal Journey with Martin Scorsese Throug...  movie   
10658                                        Berlin 1945  movie   
12065                                          Last Call  movie   
12188                                          Last Call  movie   
13091                The Kidnap & Murder of Lynda Spence  movie   
13093                The Kidnap & Murder of Lynda Spence  movie   
13337                                   De olhos abertos  movie   
13564                                   De olhos abertos  movie   
13748        The Life and Murder of Nicole Brown Simpson  movie   
13980  A Personal Journey with Martin Scorsese Throug...  movie   
16632                                        Berlin 1945     tv   
18075        The Life and Murder of Nicole Brown Simpson     tv   

                       Combined Genres  Release Year     IMDb ID  \
3497   Biography, Documentary, History          1995   tt0112120   
10658        Documentary, History, War          2020  tt12264166   
12065                      Documentary          2021         NaN   
12188                              NaN          2021         NaN   
13091               Documentary, Crime          2023         NaN   
13093               Documentary, Crime          2023         NaN   
13337                        Animation          2023         NaN   
13564                       Drama, War          2023         NaN   
13748    Biography, Crime, Documentary          2024  tt32267612   
13980  Biography, Documentary, History          1995   tt0112120   
16632        Documentary, History, War          2020  tt12264166   
18075    Biography, Crime, Documentary          2024  tt32267612   

       IMDb Average Rating       IMDb Num Votes Service Name  
3497                   8.5                 5196      AppleTV  
10658                  8.2                  352      AppleTV  
12065                  NaN -9223372036854775808      AppleTV  
12188                  NaN -9223372036854775808      AppleTV  
13091                  NaN -9223372036854775808      AppleTV  
13093                  NaN -9223372036854775808      AppleTV  
13337                  NaN -9223372036854775808      AppleTV  
13564                  NaN -9223372036854775808      AppleTV  
13748                  7.4                  191      AppleTV  
13980                  8.5                 5196      AppleTV  
16632                  8.2                  352      AppleTV  
18075                  7.4                  191      AppleTV

In [16]:
apple = apple.drop_duplicates(subset=["IMDb ID","Title","Release Year"], keep="first")
rows = apple[apple["Title"] == "The Life and Murder of Nicole Brown Simpson"]
display(rows)

Title   Type  \
13748  The Life and Murder of Nicole Brown Simpson  movie   

                     Combined Genres  Release Year     IMDb ID  \
13748  Biography, Crime, Documentary          2024  tt32267612   

       IMDb Average Rating  IMDb Num Votes Service Name  
13748                  7.4             191      AppleTV

In [17]:
#apple.genres.value_counts()
# apple['genres'].str.split(',', expand=True)
apple[["Genre 1","Genre 2","Genre 3","Genre 4","Genre 5","Genre 6"]] = apple["Combined Genres"].str.split(',',expand=True)

display(apple)



Title   Type     Combined Genres  \
0                                Four Rooms  movie              Comedy   
1                              Forrest Gump  movie      Drama, Romance   
2                           American Beauty  movie               Drama   
3                              Citizen Kane  movie      Drama, Mystery   
4                                Metropolis  movie       Drama, Sci-Fi   
...                                     ...    ...                 ...   
18188     Das Boot - Die komplette TV-Serie     tv          Drama, War   
18194              The Amazon Review Killer     tv         Documentary   
18198                          Conspirators     tv             Mystery   
18200                   Extraordinary World     tv         Documentary   
18203  They Took Our Child: We Got Her Back     tv  Crime, Documentary   

       Release Year     IMDb ID  IMDb Average Rating       IMDb Num Votes  \
0              1995   tt0113101                  6.7               113403   
1              1994   tt0109830                  8.8              2348885   
2              1999   tt0169547                  8.3              1238903   
3              1941   tt0033467                  8.3               477481   
4              1927   tt0017136                  8.3               192285   
...             ...         ...                  ...                  ...   
18188          1985  tt30970892                  8.8                  198   
18194          2024  tt33342317                  6.4                   64   
18198          2025  tt35075008                  5.5                   44   
18200          2025  tt34888944                  NaN -9223372036854775808   
18203          2015   tt5056408                  8.6                   31   

      Service Name      Genre 1       Genre 2 Genre 3 Genre 4 Genre 5 Genre 6  
0          AppleTV       Comedy          None    None    None    None    None  
1          AppleTV        Drama       Romance    None    None    None    None  
2          AppleTV        Drama          None    None    None    None    None  
3          AppleTV        Drama       Mystery    None    None    None    None  
4          AppleTV        Drama        Sci-Fi    None    None    None    None  
...            ...          ...           ...     ...     ...     ...     ...  
18188      AppleTV        Drama           War    None    None    None    None  
18194      AppleTV  Documentary          None    None    None    None    None  
18198      AppleTV      Mystery          None    None    None    None    None  
18200      AppleTV  Documentary          None    None    None    None    None  
18203      AppleTV        Crime   Documentary    None    None    None    None  

[17637 rows x 14 columns]

### **Hulu**

In [18]:
# View top 5 rows in the hulu df
hulu.head()

title   type                     genres  releaseYear  \
0                Ariel  movie     Comedy, Crime, Romance       1988.0   
1  Shadows in Paradise  movie       Comedy, Drama, Music       1986.0   
2         Forrest Gump  movie             Drama, Romance       1994.0   
3    The Fifth Element  movie  Action, Adventure, Sci-Fi       1997.0   
4   My Life Without Me  movie             Drama, Romance       2003.0   

      imdbId  imdbAverageRating  imdbNumVotes availableCountries  
0  tt0094675                7.4        8949.0                NaN  
1  tt0092149                7.5        7727.0                NaN  
2  tt0109830                8.8     2348885.0                NaN  
3  tt0119116                7.6      521815.0                NaN  
4  tt0314412                7.4       26140.0                NaN

In [19]:
hulu.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10203 entries, 0 to 10202
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   title               9552 non-null   object 
 1   type                10203 non-null  object 
 2   genres              9853 non-null   object 
 3   releaseYear         10166 non-null  float64
 4   imdbId              9138 non-null   object 
 5   imdbAverageRating   8830 non-null   float64
 6   imdbNumVotes        8830 non-null   float64
 7   availableCountries  44 non-null     object 
dtypes: float64(3), object(5)
memory usage: 637.8+ KB


In [20]:
hulu.describe()

releaseYear  imdbAverageRating  imdbNumVotes
count  10166.000000        8830.000000  8.830000e+03
mean    2011.744344           6.569921  3.536248e+04
std       13.952201           1.055329  1.456587e+05
min     1929.000000           1.800000  5.000000e+00
25%     2009.000000           6.000000  1.390000e+02
50%     2016.000000           6.700000  9.210000e+02
75%     2021.000000           7.300000  7.385750e+03
max     2025.000000           9.500000  3.002274e+06

In [21]:
hulu.select_dtypes("object").describe()

title   type genres     imdbId availableCountries
count   9552  10203   9853       9138                 44
unique  9342      2    664       9134                  3
top     Prey  movie  Drama  tt1945851                 JP
freq       4   6019   1031          2                 28

In [22]:
# Drop availableCountries Column - no value with such a low non-null number
hulu.drop("availableCountries",axis=1,inplace=True)
hulu.columns

Index(['title', 'type', 'genres', 'releaseYear', 'imdbId', 'imdbAverageRating',
       'imdbNumVotes'],
      dtype='object')

In [23]:
# Change releaseYear,  imdbNumVotes to integers 
# Display only the data types

hulu[["releaseYear","imdbNumVotes"]] = hulu[["releaseYear","imdbNumVotes"]].apply(np.int64)
display(hulu.dtypes)

c:\Users\rkynhoff\OneDrive - University of Rochester\Documents\CodeLouisville\Streaming_Service_Comparisons\venv\Lib\site-packages\pandas\core\series.py:1031: RuntimeWarning: invalid value encountered in cast
  arr = np.asarray(values, dtype=dtype)


title                 object
type                  object
genres                object
releaseYear            int64
imdbId                object
imdbAverageRating    float64
imdbNumVotes           int64
dtype: object

In [24]:
# Rename Columns
hulu.rename(columns={"title":"Title","type":"Type","genres":"Combined Genres","releaseYear":"Release Year","imdbId":"IMDb ID","imdbAverageRating":"IMDb Average Rating","imdbNumVotes":"IMDb Num Votes"},inplace = True)
# Add column that identifies the streaming service
hulu["Service Name"] = "Hulu"
hulu.columns

Index(['Title', 'Type', 'Combined Genres', 'Release Year', 'IMDb ID',
       'IMDb Average Rating', 'IMDb Num Votes', 'Service Name'],
      dtype='object')

In [25]:
# Determine how many titles contain null values
hulu["Title"].isna().sum()

np.int64(651)

In [26]:
# Drop the rows contiaining null values in the Title column
hulu = hulu.dropna(subset=["Title"])
hulu["Title"].isna().sum()

np.int64(0)

In [27]:
hulu.duplicated().value_counts()

False    9550
True        2
Name: count, dtype: int64

In [28]:
hulu[hulu.duplicated(keep=False)]

Title Type     Combined Genres  Release Year    IMDb ID  \
6811       The Gutsy Frog   tv              Comedy          2015  tt4658636   
7193  Scorned: Love Kills   tv  Crime, Documentary          2012  tt2287041   
7194  Scorned: Love Kills   tv  Crime, Documentary          2012  tt2287041   
8646       The Gutsy Frog   tv              Comedy          2015  tt4658636   

      IMDb Average Rating  IMDb Num Votes Service Name  
6811                  7.1              17         Hulu  
7193                  6.9             304         Hulu  
7194                  6.9             304         Hulu  
8646                  7.1              17         Hulu

In [29]:
hulu.drop_duplicates()


Title   Type  \
0                                     Ariel  movie   
1                       Shadows in Paradise  movie   
2                              Forrest Gump  movie   
3                         The Fifth Element  movie   
4                        My Life Without Me  movie   
...                                     ...    ...   
10194                          Scam Goddess     tv   
10196             AQUARION Myth of Emotions     tv   
10199  They Took Our Child: We Got Her Back     tv   
10201     Cardcaptor Sakura: Clear Card Arc     tv   
10202                        Faces of Music     tv   

                    Combined Genres  Release Year     IMDb ID  \
0            Comedy, Crime, Romance          1988   tt0094675   
1              Comedy, Drama, Music          1986   tt0092149   
2                    Drama, Romance          1994   tt0109830   
3         Action, Adventure, Sci-Fi          1997   tt0119116   
4                    Drama, Romance          2003   tt0314412   
...                             ...           ...         ...   
10194              Crime, Talk-Show          2025  tt32821425   
10196    Action, Animation, Fantasy          2024  tt30644518   
10199            Crime, Documentary          2015   tt5056408   
10201  Adventure, Animation, Comedy          2018   tt6279576   
10202                   Documentary          2025  tt35460841   

       IMDb Average Rating       IMDb Num Votes Service Name  
0                      7.4                 8949         Hulu  
1                      7.5                 7727         Hulu  
2                      8.8              2348885         Hulu  
3                      7.6               521815         Hulu  
4                      7.4                26140         Hulu  
...                    ...                  ...          ...  
10194                  6.7                   58         Hulu  
10196                  3.5                   17         Hulu  
10199                  8.6                   31         Hulu  
10201                  7.7                  879         Hulu  
10202                  NaN -9223372036854775808         Hulu  

[9550 rows x 8 columns]

In [30]:
hulu[hulu.duplicated(subset=["IMDb ID","Title","Release Year"], keep=False)]


Title   Type             Combined Genres  Release Year  \
2608  Panzer World Galient  movie  Action, Animation, Fantasy          1984   
5480    Shylock's Children  movie                Crime, Drama          2023   
6811        The Gutsy Frog     tv                      Comedy          2015   
7193   Scorned: Love Kills     tv          Crime, Documentary          2012   
7194   Scorned: Love Kills     tv          Crime, Documentary          2012   
8110  Panzer World Galient     tv  Action, Animation, Fantasy          1984   
8646        The Gutsy Frog     tv                      Comedy          2015   
9339    Shylock's Children     tv                Crime, Drama          2023   

         IMDb ID  IMDb Average Rating  IMDb Num Votes Service Name  
2608   tt1945851                  7.4              28         Hulu  
5480  tt21944604                  6.4             127         Hulu  
6811   tt4658636                  7.1              17         Hulu  
7193   tt2287041                  6.9             304         Hulu  
7194   tt2287041                  6.9             304         Hulu  
8110   tt1945851                  7.4              28         Hulu  
8646   tt4658636                  7.1              17         Hulu  
9339  tt21944604                  6.4             127         Hulu

In [31]:
hulu = hulu.drop_duplicates(subset=["IMDb ID","Title","Release Year"], keep="first")
rows = hulu[hulu["Title"] == "Panzer World Galient"]
display(rows)

Title   Type             Combined Genres  Release Year  \
2608  Panzer World Galient  movie  Action, Animation, Fantasy          1984   

        IMDb ID  IMDb Average Rating  IMDb Num Votes Service Name  
2608  tt1945851                  7.4              28         Hulu

In [32]:
#Obtain max number of elements in the Genres column
hulu["Combined Genres"].str.split(", ",expand=True)


0            1        2     3     4     5     6
0           Comedy        Crime  Romance  None  None  None  None
1           Comedy        Drama    Music  None  None  None  None
2            Drama      Romance     None  None  None  None  None
3           Action    Adventure   Sci-Fi  None  None  None  None
4            Drama      Romance     None  None  None  None  None
...            ...          ...      ...   ...   ...   ...   ...
10194        Crime    Talk-Show     None  None  None  None  None
10196       Action    Animation  Fantasy  None  None  None  None
10199        Crime  Documentary     None  None  None  None  None
10201    Adventure    Animation   Comedy  None  None  None  None
10202  Documentary         None     None  None  None  None  None

[9548 rows x 7 columns]

In [33]:
# Split the elements in the Genres column while creating new columns for the max number 
hulu[["Genre 1","Genre 2","Genre 3","Genre 4","Genre 5","Genre 6","Genre 7"]] = hulu["Combined Genres"].str.split(',',expand=True)

hulu.head(2)

Title   Type         Combined Genres  Release Year  \
0                Ariel  movie  Comedy, Crime, Romance          1988   
1  Shadows in Paradise  movie    Comedy, Drama, Music          1986   

     IMDb ID  IMDb Average Rating  IMDb Num Votes Service Name Genre 1  \
0  tt0094675                  7.4            8949         Hulu  Comedy   
1  tt0092149                  7.5            7727         Hulu  Comedy   

  Genre 2   Genre 3 Genre 4 Genre 5 Genre 6 Genre 7  
0   Crime   Romance    None    None    None    None  
1   Drama     Music    None    None    None    None

### **Netflix**

In [34]:
# View top 5 rows in the netflix df
netflix.head()

title   type                     genres  releaseYear  \
0    American Beauty  movie                      Drama       1999.0   
1  The Fifth Element  movie  Action, Adventure, Sci-Fi       1997.0   
2  Kill Bill: Vol. 1  movie    Action, Crime, Thriller       2003.0   
3            Jarhead  movie      Biography, Drama, War       2005.0   
4         Unforgiven  movie             Drama, Western       1992.0   

      imdbId  imdbAverageRating  imdbNumVotes availableCountries  
0  tt0169547                8.3     1238903.0                NaN  
1  tt0119116                7.6      521815.0                NaN  
2  tt0266697                8.2     1236086.0                NaN  
3  tt0418763                7.0      213667.0                NaN  
4  tt0105695                8.2      448833.0                NaN

In [35]:
netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20736 entries, 0 to 20735
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   title               20107 non-null  object 
 1   type                20736 non-null  object 
 2   genres              20391 non-null  object 
 3   releaseYear         20702 non-null  float64
 4   imdbId              19238 non-null  object 
 5   imdbAverageRating   19049 non-null  float64
 6   imdbNumVotes        19049 non-null  float64
 7   availableCountries  167 non-null    object 
dtypes: float64(3), object(5)
memory usage: 1.3+ MB


In [36]:
netflix.describe()

releaseYear  imdbAverageRating  imdbNumVotes
count  20702.000000       19049.000000  1.904900e+04
mean    2013.218288           6.397438  3.166172e+04
std       14.337227           1.096381  1.212741e+05
min     1913.000000           1.200000  5.000000e+00
25%     2012.000000           5.700000  3.250000e+02
50%     2018.000000           6.500000  1.572000e+03
75%     2022.000000           7.200000  1.000600e+04
max     2025.000000           9.800000  3.002274e+06

In [37]:
netflix.select_dtypes("object").describe()

title   type  genres     imdbId  \
count               20107  20736   20391      19238   
unique              19334      2     855      19235   
top     Perfect Strangers  movie  Comedy  tt6446792   
freq                    5  15762    1754          2   

                                       availableCountries  
count                                                 167  
unique                                                103  
top     AD, AE, AG, AL, AO, AR, AT, AU, AZ, BA, BB, BE...  
freq                                                   11

In [38]:
# Drop availableCountries Column - no value with such a low non-null number
netflix.drop("availableCountries",axis=1,inplace=True)
netflix.columns


Index(['title', 'type', 'genres', 'releaseYear', 'imdbId', 'imdbAverageRating',
       'imdbNumVotes'],
      dtype='object')

In [39]:
# Change releaseYear,  imdbNumVotes to integers 
# Display only the data types

netflix[["releaseYear","imdbNumVotes"]] = netflix[["releaseYear","imdbNumVotes"]].apply(np.int64)
display(netflix.dtypes)


c:\Users\rkynhoff\OneDrive - University of Rochester\Documents\CodeLouisville\Streaming_Service_Comparisons\venv\Lib\site-packages\pandas\core\series.py:1031: RuntimeWarning: invalid value encountered in cast
  arr = np.asarray(values, dtype=dtype)


title                 object
type                  object
genres                object
releaseYear            int64
imdbId                object
imdbAverageRating    float64
imdbNumVotes           int64
dtype: object

In [40]:
# Rename Columns
netflix.rename(columns={"title":"Title","type":"Type","genres":"Combined Genres","releaseYear":"Release Year","imdbId":"IMDb ID","imdbAverageRating":"IMDb Average Rating","imdbNumVotes":"IMDb Num Votes"},inplace = True)
# Add column that identifies the streaming service
netflix["Service Name"] = "Netflix"
netflix.columns


Index(['Title', 'Type', 'Combined Genres', 'Release Year', 'IMDb ID',
       'IMDb Average Rating', 'IMDb Num Votes', 'Service Name'],
      dtype='object')

In [41]:
# Determine how many titles contain null values
netflix["Title"].isna().sum()


np.int64(629)

In [42]:
# Drop the rows contiaining null values in the Title column
netflix = netflix.dropna(subset=["Title"])
netflix["Title"].isna().sum()


np.int64(0)

In [43]:
netflix.duplicated().value_counts()

False    20107
Name: count, dtype: int64

In [44]:
netflix[netflix.duplicated(keep=False)]

Empty DataFrame
Columns: [Title, Type, Combined Genres, Release Year, IMDb ID, IMDb Average Rating, IMDb Num Votes, Service Name]
Index: []

In [45]:
netflix.drop_duplicates()

Title   Type               Combined Genres  \
0                     American Beauty  movie                         Drama   
1                   The Fifth Element  movie     Action, Adventure, Sci-Fi   
2                   Kill Bill: Vol. 1  movie       Action, Crime, Thriller   
3                             Jarhead  movie         Biography, Drama, War   
4                          Unforgiven  movie                Drama, Western   
...                               ...    ...                           ...   
20724         Grisaia Phantom Trigger     tv      Action, Animation, Drama   
20725     Legend of Princess Chang-Ge     tv  Action, Adventure, Animation   
20733                      Ms. Rachel     tv                        Family   
20734  American Manhunt: O.J. Simpson     tv                   Documentary   
20735                   Devil's Diner     tv                 Drama, Horror   

       Release Year     IMDb ID  IMDb Average Rating       IMDb Num Votes  \
0              1999   tt0169547                  8.3              1238903   
1              1997   tt0119116                  7.6               521815   
2              2003   tt0266697                  8.2              1236086   
3              2005   tt0418763                  7.0               213667   
4              1992   tt0105695                  8.2               448833   
...             ...         ...                  ...                  ...   
20724          2025  tt33503859                  4.7                   32   
20725          2024  tt32561196                  NaN -9223372036854775808   
20733          2025  tt35406274                  8.0                   23   
20734          2025  tt35456246                  7.4                 1148   
20735          2025  tt35557166                  7.2                  289   

      Service Name  
0          Netflix  
1          Netflix  
2          Netflix  
3          Netflix  
4          Netflix  
...            ...  
20724      Netflix  
20725      Netflix  
20733      Netflix  
20734      Netflix  
20735      Netflix  

[20107 rows x 8 columns]

In [46]:
netflix[netflix.duplicated(subset=["IMDb ID","Title","Release Year"], keep=False)]

Title   Type               Combined Genres  \
6318    Gekijouban Yowamushi pedaru  movie    Animation, Comedy, Romance   
7219    Miniforce - New Heroes Rise  movie  Action, Adventure, Animation   
9350        The Best of the Wiggles  movie                 Music, Family   
10361                         Lucid  movie                           NaN   
10441  True and the Rainbow Kingdom  movie  Adventure, Animation, Comedy   
10874              Amr's in Trouble  movie               Romance, Comedy   
11217                         Lucid  movie                         Drama   
13522       The Best of the Wiggles  movie                           NaN   
14327              Amr's in Trouble  movie                           NaN   
14945              De olhos abertos  movie                     Animation   
15247              De olhos abertos  movie                    Drama, War   
16929   Miniforce - New Heroes Rise     tv  Action, Adventure, Animation   
16977  True and the Rainbow Kingdom     tv  Adventure, Animation, Comedy   
18606   Gekijouban Yowamushi pedaru     tv    Animation, Comedy, Romance   

       Release Year    IMDb ID  IMDb Average Rating       IMDb Num Votes  \
6318           2015  tt4982154                  6.7                  160   
7219           2016  tt6446792                  6.1                   55   
9350           2018        NaN                  NaN -9223372036854775808   
10361          2019        NaN                  NaN -9223372036854775808   
10441          2017  tt5607658                  7.0                  574   
10874          2019        NaN                  NaN -9223372036854775808   
11217          2019        NaN                  NaN -9223372036854775808   
13522          2018        NaN                  NaN -9223372036854775808   
14327          2019        NaN                  NaN -9223372036854775808   
14945          2023        NaN                  NaN -9223372036854775808   
15247          2023        NaN                  NaN -9223372036854775808   
16929          2016  tt6446792                  6.1                   55   
16977          2017  tt5607658                  7.0                  574   
18606          2015  tt4982154                  6.7                  160   

      Service Name  
6318       Netflix  
7219       Netflix  
9350       Netflix  
10361      Netflix  
10441      Netflix  
10874      Netflix  
11217      Netflix  
13522      Netflix  
14327      Netflix  
14945      Netflix  
15247      Netflix  
16929      Netflix  
16977      Netflix  
18606      Netflix

In [47]:
netflix = netflix.drop_duplicates(subset=["IMDb ID","Title","Release Year"], keep="first")
rows = netflix[netflix["Title"] == "Gekijouban Yowamushi pedaru"]
display(rows)

Title   Type             Combined Genres  \
6318  Gekijouban Yowamushi pedaru  movie  Animation, Comedy, Romance   

      Release Year    IMDb ID  IMDb Average Rating  IMDb Num Votes  \
6318          2015  tt4982154                  6.7             160   

     Service Name  
6318      Netflix

In [48]:
#Obtain max number of elements in the Genres column
netflix["Combined Genres"].str.split(", ",expand=True)


0          1          2     3     4
0            Drama       None       None  None  None
1           Action  Adventure     Sci-Fi  None  None
2           Action      Crime   Thriller  None  None
3        Biography      Drama        War  None  None
4            Drama    Western       None  None  None
...            ...        ...        ...   ...   ...
20724       Action  Animation      Drama  None  None
20725       Action  Adventure  Animation  None  None
20733       Family       None       None  None  None
20734  Documentary       None       None  None  None
20735        Drama     Horror       None  None  None

[20100 rows x 5 columns]

In [49]:
# Split the elements in the Genres column while creating new columns for the max number 
netflix[["Genre 1","Genre 2","Genre 3","Genre 4","Genre 5"]] = netflix["Combined Genres"].str.split(',',expand=True)

netflix.head(2)


Title   Type            Combined Genres  Release Year  \
0    American Beauty  movie                      Drama          1999   
1  The Fifth Element  movie  Action, Adventure, Sci-Fi          1997   

     IMDb ID  IMDb Average Rating  IMDb Num Votes Service Name Genre 1  \
0  tt0169547                  8.3         1238903      Netflix   Drama   
1  tt0119116                  7.6          521815      Netflix  Action   

      Genre 2  Genre 3 Genre 4 Genre 5  
0        None     None    None    None  
1   Adventure   Sci-Fi    None    None

### **Prime**

In [50]:
# View top 5 rows of prime df
prime.head()

title   type                  genres  releaseYear     imdbId  \
0           Ariel  movie  Comedy, Crime, Romance       1988.0  tt0094675   
1      Four Rooms  movie                  Comedy       1995.0  tt0113101   
2  Judgment Night  movie    Action, Crime, Drama       1993.0  tt0107286   
3    Forrest Gump  movie          Drama, Romance       1994.0  tt0109830   
4    Citizen Kane  movie          Drama, Mystery       1941.0  tt0033467   

   imdbAverageRating  imdbNumVotes availableCountries  
0                7.4        8949.0                NaN  
1                6.7      113403.0                NaN  
2                6.6       19627.0                NaN  
3                8.8     2348885.0                NaN  
4                8.3      477481.0                NaN

In [51]:
prime.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70055 entries, 0 to 70054
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   title               68386 non-null  object 
 1   type                70055 non-null  object 
 2   genres              67939 non-null  object 
 3   releaseYear         69868 non-null  float64
 4   imdbId              63648 non-null  object 
 5   imdbAverageRating   61316 non-null  float64
 6   imdbNumVotes        61316 non-null  float64
 7   availableCountries  373 non-null    object 
dtypes: float64(3), object(5)
memory usage: 4.3+ MB


In [52]:
prime.describe()

releaseYear  imdbAverageRating  imdbNumVotes
count  69868.000000       61316.000000  6.131600e+04
mean    2007.542494           5.955402  1.324737e+04
std       18.939153           1.370233  7.553937e+04
min     1902.000000           1.000000  5.000000e+00
25%     2004.000000           5.100000  1.010000e+02
50%     2015.000000           6.100000  4.460000e+02
75%     2019.000000           7.000000  2.331000e+03
max     2026.000000          10.000000  3.002274e+06

In [53]:
prime.select_dtypes("object").describe()

title   type genres     imdbId availableCountries
count     68386  70055  67939      63648                373
unique    63384      2   1465      63625                 97
top     Monster  movie  Drama  tt0468988                 US
freq          9  60901   6743          2                110

In [54]:
# Drop availableCountries Column - no value with such a low non-null number
prime.drop("availableCountries",axis=1,inplace=True)
prime.columns


Index(['title', 'type', 'genres', 'releaseYear', 'imdbId', 'imdbAverageRating',
       'imdbNumVotes'],
      dtype='object')

In [55]:
# Change releaseYear,  imdbNumVotes to integers 
# Display only the data types
prime[["releaseYear","imdbNumVotes"]] = prime[["releaseYear","imdbNumVotes"]].apply(np.int64)
display(prime.dtypes)


c:\Users\rkynhoff\OneDrive - University of Rochester\Documents\CodeLouisville\Streaming_Service_Comparisons\venv\Lib\site-packages\pandas\core\series.py:1031: RuntimeWarning: invalid value encountered in cast
  arr = np.asarray(values, dtype=dtype)


title                 object
type                  object
genres                object
releaseYear            int64
imdbId                object
imdbAverageRating    float64
imdbNumVotes           int64
dtype: object

In [56]:
# Rename Columns
prime.rename(columns={"title":"Title","type":"Type","genres":"Combined Genres","releaseYear":"Release Year","imdbId":"IMDb ID","imdbAverageRating":"IMDb Average Rating","imdbNumVotes":"IMDb Num Votes"},inplace = True)
# Add column that identifies the streaming service
prime["Service Name"] = "Prime"
prime.columns


Index(['Title', 'Type', 'Combined Genres', 'Release Year', 'IMDb ID',
       'IMDb Average Rating', 'IMDb Num Votes', 'Service Name'],
      dtype='object')

In [57]:
# # Add a column to the prime df and entered a default value for each row & display bottom two rows
# prime["Service Name"] = "Prime" - done above, but keeping notes below for future function building
# prime.tail(2) 

# map function for creating column and populating each cell within each df with the df name

# create a dictionary with the list of the names of df and have it look at the df name to populate the column

In [58]:
# Determine how many titles contain null values
prime["Title"].isna().sum()


np.int64(1669)

In [59]:
# Drop the rows contiaining null values in the Title column
prime = prime.dropna(subset=["Title"])
prime["Title"].isna().sum()


np.int64(0)

In [60]:
prime.duplicated().value_counts()

False    68382
True         4
Name: count, dtype: int64

In [61]:
prime[prime.duplicated(keep=False)]

Title   Type     Combined Genres  Release Year  \
25185                 Burn  movie                 NaN          2014   
35993                Belle  movie     Fantasy, Horror          2023   
42922                 Burn  movie                 NaN          2014   
45350                Belle  movie     Fantasy, Horror          2023   
50542           Flower Boy  movie                 NaN          2021   
51337           Flower Boy  movie                 NaN          2021   
63744  Scorned: Love Kills     tv  Crime, Documentary          2012   
63745  Scorned: Love Kills     tv  Crime, Documentary          2012   

          IMDb ID  IMDb Average Rating       IMDb Num Votes Service Name  
25185         NaN                  NaN -9223372036854775808        Prime  
35993  tt12373754                  5.4                  168        Prime  
42922         NaN                  NaN -9223372036854775808        Prime  
45350  tt12373754                  5.4                  168        Prime  
50542         NaN                  NaN -9223372036854775808        Prime  
51337         NaN                  NaN -9223372036854775808        Prime  
63744   tt2287041                  6.9                  304        Prime  
63745   tt2287041                  6.9                  304        Prime

In [62]:
prime.drop_duplicates()

Title   Type         Combined Genres  Release Year  \
0                    Ariel  movie  Comedy, Crime, Romance          1988   
1               Four Rooms  movie                  Comedy          1995   
2           Judgment Night  movie    Action, Crime, Drama          1993   
3             Forrest Gump  movie          Drama, Romance          1994   
4             Citizen Kane  movie          Drama, Mystery          1941   
...                    ...    ...                     ...           ...   
70046  Breaking Pickleball     tv             Documentary          2024   
70050         Abracadavers     tv  Action, Comedy, Sci-Fi          2019   
70051    Tom Green Country     tv                 Reality          2025   
70053          Ryan Trahan     tv              Reality-TV          2019   
70054          Sivarapalli     tv                  Comedy          2025   

          IMDb ID  IMDb Average Rating  IMDb Num Votes Service Name  
0       tt0094675                  7.4            8949        Prime  
1       tt0113101                  6.7          113403        Prime  
2       tt0107286                  6.6           19627        Prime  
3       tt0109830                  8.8         2348885        Prime  
4       tt0033467                  8.3          477481        Prime  
...           ...                  ...             ...          ...  
70046  tt32306198                  9.3              76        Prime  
70050   tt8386640                  8.2              40        Prime  
70051  tt32832543                  8.1              72        Prime  
70053  tt28487606                  8.2              42        Prime  
70054  tt31914057                  7.4             120        Prime  

[68382 rows x 8 columns]

In [63]:
prime[prime.duplicated(subset=["IMDb ID","Title","Release Year"], keep=False)]

Title   Type            Combined Genres  \
830                  Moses the Lawgiver  movie                      Drama   
3339                            Ferrari  movie                      Drama   
6850                               Icon  movie    Action, Crime, Thriller   
7782             Gone But Not Forgotten  movie      Crime, Drama, Mystery   
7809   Category 7: The End of the World  movie   Action, Adventure, Drama   
...                                 ...    ...                        ...   
68195            Gone But Not Forgotten     tv      Crime, Drama, Mystery   
68300                Shylock's Children     tv               Crime, Drama   
68510                        Love Kills     tv                     Horror   
68609              Living with the Dead     tv      Crime, Drama, Mystery   
69976                 Atatürk 1881-1919     tv  Biography, Drama, History   

       Release Year     IMDb ID  IMDb Average Rating  IMDb Num Votes  \
830            1974   tt0072547                  6.1             591   
3339           2003   tt0346946                  7.0             925   
6850           2005   tt0424176                  5.6            1085   
7782           2005   tt0384736                  6.1             732   
7809           2005   tt0468988                  4.5            3044   
...             ...         ...                  ...             ...   
68195          2005   tt0384736                  6.1             732   
68300          2023  tt21944604                  6.4             127   
68510          2023  tt16431352                  6.3              11   
68609          2002   tt0289652                  7.0            1496   
69976          2024  tt15676170                  8.8            7474   

      Service Name  
830          Prime  
3339         Prime  
6850         Prime  
7782         Prime  
7809         Prime  
...            ...  
68195        Prime  
68300        Prime  
68510        Prime  
68609        Prime  
69976        Prime  

[74 rows x 8 columns]

In [64]:
prime[prime.duplicated(subset=["IMDb ID","Title","Release Year"], keep=False)]


Title   Type            Combined Genres  \
830                  Moses the Lawgiver  movie                      Drama   
3339                            Ferrari  movie                      Drama   
6850                               Icon  movie    Action, Crime, Thriller   
7782             Gone But Not Forgotten  movie      Crime, Drama, Mystery   
7809   Category 7: The End of the World  movie   Action, Adventure, Drama   
...                                 ...    ...                        ...   
68195            Gone But Not Forgotten     tv      Crime, Drama, Mystery   
68300                Shylock's Children     tv               Crime, Drama   
68510                        Love Kills     tv                     Horror   
68609              Living with the Dead     tv      Crime, Drama, Mystery   
69976                 Atatürk 1881-1919     tv  Biography, Drama, History   

       Release Year     IMDb ID  IMDb Average Rating  IMDb Num Votes  \
830            1974   tt0072547                  6.1             591   
3339           2003   tt0346946                  7.0             925   
6850           2005   tt0424176                  5.6            1085   
7782           2005   tt0384736                  6.1             732   
7809           2005   tt0468988                  4.5            3044   
...             ...         ...                  ...             ...   
68195          2005   tt0384736                  6.1             732   
68300          2023  tt21944604                  6.4             127   
68510          2023  tt16431352                  6.3              11   
68609          2002   tt0289652                  7.0            1496   
69976          2024  tt15676170                  8.8            7474   

      Service Name  
830          Prime  
3339         Prime  
6850         Prime  
7782         Prime  
7809         Prime  
...            ...  
68195        Prime  
68300        Prime  
68510        Prime  
68609        Prime  
69976        Prime  

[74 rows x 8 columns]

In [65]:
# prime = prime.drop_duplicates(subset=["IMDb ID","Title","Release Year"], keep="first")
# rows = prime[prime["Title"] == "Panzer World Galient"]
# display(rows)

In [66]:
#Obtain max number of elements in the Genres column
prime["Combined Genres"].str.split(", ",expand=True)


0        1        2     3     4     5
0           Comedy    Crime  Romance  None  None  None
1           Comedy     None     None  None  None  None
2           Action    Crime    Drama  None  None  None
3            Drama  Romance     None  None  None  None
4            Drama  Mystery     None  None  None  None
...            ...      ...      ...   ...   ...   ...
70046  Documentary     None     None  None  None  None
70050       Action   Comedy   Sci-Fi  None  None  None
70051      Reality     None     None  None  None  None
70053   Reality-TV     None     None  None  None  None
70054       Comedy     None     None  None  None  None

[68386 rows x 6 columns]

In [67]:
# Split the elements in the Genres column while creating new columns for the max number 
prime[["Genre 1","Genre 2","Genre 3","Genre 4","Genre 5","Genre 6"]] = prime["Combined Genres"].str.split(',',expand=True)
prime.head(2)

Title   Type         Combined Genres  Release Year    IMDb ID  \
0       Ariel  movie  Comedy, Crime, Romance          1988  tt0094675   
1  Four Rooms  movie                  Comedy          1995  tt0113101   

   IMDb Average Rating  IMDb Num Votes Service Name Genre 1 Genre 2   Genre 3  \
0                  7.4            8949        Prime  Comedy   Crime   Romance   
1                  6.7          113403        Prime  Comedy    None      None   

  Genre 4 Genre 5 Genre 6  
0    None    None    None  
1    None    None    None

In [68]:
# Merge the dataframes to append them to the end of each other since using the same column names
streaming = pd.concat([apple, hulu, netflix, prime], axis = 0)


In [69]:
streaming.info()

<class 'pandas.core.frame.DataFrame'>
Index: 115671 entries, 0 to 70054
Data columns (total 15 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   Title                115671 non-null  object 
 1   Type                 115671 non-null  object 
 2   Combined Genres      113456 non-null  object 
 3   Release Year         115671 non-null  int64  
 4   IMDb ID              108741 non-null  object 
 5   IMDb Average Rating  105499 non-null  float64
 6   IMDb Num Votes       115671 non-null  int64  
 7   Service Name         115671 non-null  object 
 8   Genre 1              113456 non-null  object 
 9   Genre 2              75644 non-null   object 
 10  Genre 3              44152 non-null   object 
 11  Genre 4              117 non-null     object 
 12  Genre 5              26 non-null      object 
 13  Genre 6              4 non-null       object 
 14  Genre 7              1 non-null       object 
dtypes: float64(1), int64(2)

In [70]:
streaming.describe()

Release Year  IMDb Average Rating  IMDb Num Votes
count  1.156710e+05        105499.000000    1.156710e+05
mean  -7.335894e+15             6.152437   -8.110947e+17
std    2.600163e+17             1.290911    2.612128e+18
min   -9.223372e+18             1.000000   -9.223372e+18
25%    2.005000e+03             5.400000    7.800000e+01
50%    2.015000e+03             6.300000    5.040000e+02
75%    2.020000e+03             7.100000    3.266000e+03
max    2.026000e+03            10.000000    3.002274e+06

In [71]:
streaming.duplicated().value_counts()

False    115667
True          4
Name: count, dtype: int64

In [72]:
streaming[streaming.duplicated(keep=False)]

Title   Type     Combined Genres  Release Year  \
25185                 Burn  movie                 NaN          2014   
35993                Belle  movie     Fantasy, Horror          2023   
42922                 Burn  movie                 NaN          2014   
45350                Belle  movie     Fantasy, Horror          2023   
50542           Flower Boy  movie                 NaN          2021   
51337           Flower Boy  movie                 NaN          2021   
63744  Scorned: Love Kills     tv  Crime, Documentary          2012   
63745  Scorned: Love Kills     tv  Crime, Documentary          2012   

          IMDb ID  IMDb Average Rating       IMDb Num Votes Service Name  \
25185         NaN                  NaN -9223372036854775808        Prime   
35993  tt12373754                  5.4                  168        Prime   
42922         NaN                  NaN -9223372036854775808        Prime   
45350  tt12373754                  5.4                  168        Prime   
50542         NaN                  NaN -9223372036854775808        Prime   
51337         NaN                  NaN -9223372036854775808        Prime   
63744   tt2287041                  6.9                  304        Prime   
63745   tt2287041                  6.9                  304        Prime   

       Genre 1       Genre 2 Genre 3 Genre 4 Genre 5 Genre 6 Genre 7  
25185      NaN           NaN     NaN     NaN     NaN     NaN     NaN  
35993  Fantasy        Horror    None    None    None    None     NaN  
42922      NaN           NaN     NaN     NaN     NaN     NaN     NaN  
45350  Fantasy        Horror    None    None    None    None     NaN  
50542      NaN           NaN     NaN     NaN     NaN     NaN     NaN  
51337      NaN           NaN     NaN     NaN     NaN     NaN     NaN  
63744    Crime   Documentary    None    None    None    None     NaN  
63745    Crime   Documentary    None    None    None    None     NaN

In [74]:
streaming["Title"].isnull().value_counts()


Title
False    115671
Name: count, dtype: int64

In [75]:
streaming["IMDb ID"].isnull().value_counts()

IMDb ID
False    108741
True       6930
Name: count, dtype: int64

In [ ]:
# # Assign dataset names & combine to read in as separate data frames

# list_of_names = ['Netflix','Hulu','Prime','AppleTV']

# # Created an empty list into which I can place the datasets
# combined_list = []

# # Used a function to append the datasets into the empty list I created above
# for i in range(len(list_of_names)):
#     temp_df = pd.read_csv(list_of_names[i]+".csv")
#     combined_list.append(temp_df)

### **Initial Checks**

In [ ]:
# # Looking at file content of list in index 0 - Netflix - to determine column names, non-null counts, datatypes, and what the top five rows look like
# combined_list[0].info()
# combined_list[0].head(5)



In [ ]:
# # Looking at file content of list in index 1 - Hulu - to determine column names, non-null counts, datatypes, and what the top five rows look like
# combined_list[1].info()
# combined_list[1].head(5)

In [ ]:
# # Looking at file content of list in index 2 - Prime - to determine column names, non-null counts, datatypes, and what the top five rows look like
# combined_list[2].info()
# combined_list[2].head(5)

In [ ]:
# # Looking at file content of list in index 3 - AppleTV - to determine column names, non-null counts, datatypes, and what the top five rows look like
# combined_list[3].info()
# combined_list[3].head(5)

### Initiall Review (02/10/2025)
- The "availableCountries" column will not provide much data going forward
- No dataset has a column for the source of the data, so this will need to be added

### Secondary Review (02/12/2025)
- Each dataset contains an "imdbId" column with the listing's IMDb ID, which is a string; this may come in handy 
- There are 3,514 missing titles
    - Prime = 2234
    - AppleTV = 651
    - Hulu = 2885
    - Netflix = 629
- There are 2,590 duplicated rows, among which many are from missing titles, though several are duplicate title names 
    - Apple = 2
    - Hulu = 651


### Breakdown:
#### Apple
- 565 Missing Titles
- 2 Dupliate Rows
#### Hulu:
- 651 Missing Titles
- 2 Duplicated Rows